File: CS 470/Project 1/Tennis/Match.cpp
#include "Match.hpp"
#include "MatchScore.hpp"
#include "Score.hpp"
#include "Set.hpp"
#include "Score.hpp"
#include "SetScore.hpp"

Match::Match( Player *p1, Player *p2 ): Competition( p1, p2 ) {}
Score *Match::play( Player *p ) {
    MatchScore *matchScore = new MatchScore( player1(), player2() );
    while ( !matchScore->haveAWinner() ) {
        Set *set = new Set( player1(), player2() );
        Score *setScore = set->play( player1() );
        matchScore -> addScore(setScore);
        delete set;
        p = p -> otherPlayer(p);
    }
    return matchScore;
}

File: TennisInObjectiveC/TennisInObjectiveC/Tennis Files/Match.m
//
//  Match.m
//  TennisInObjectiveC
//
//  Created by Stephen Kyles on 1/31/14.
//  Copyright (c) 2014 Stephen Kyles. All rights reserved.
//

#import "Match.h"
#import "Player.h"
#import "Score.h"
#import "MatchScore.h"
#import "Set.h"

@implementation Match

-(instancetype) initWithFirstPlayer: (Player *) p1 secondPlayer:(Player *) p2
{
    if( (self = [super initWithFirstPlayer:p1 secondPlayer:p2] ) == nil )
        return nil;
    return self;
}

-(Score *) play:(Player *)player
{
    MatchScore *matchScore = [[MatchScore alloc] initWithFirstPlayer:self.player1 secondPlayer:self.player2];
    
    while( ! [matchScore haveAWinner] ) {
        Set *set = [[Set alloc] initWithFirstPlayer:self.player1 secondPlayer:self.player2];
        Score *score = [set play: player];
        [matchScore addScore: score];
        score = nil;
        player = [Player otherPlayer:player];
    }
    return matchScore;
}

@end
